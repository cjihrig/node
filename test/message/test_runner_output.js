'use strict';
require('../common');
const assert = require('assert');
const test = require('test_runner');

test('sync pass todo', (t) => {
  t.todo();
});

test('sync pass todo with message', (t) => {
  t.todo('this is a passing todo');
});

test('sync fail todo', (t) => {
  t.todo();
  throw new Error('thrown from sync fail todo');
});

test('sync fail todo with message', (t) => {
  t.todo('this is a failing todo');
  throw new Error('thrown from sync fail todo with message');
});

test('sync skip pass', (t) => {
  t.skip();
});

test('sync skip pass with message', (t) => {
  t.skip('this is skipped');
});

test('sync pass', (t) => {
  t.diagnostic('this test should pass');
});

test('sync throw fail', () => {
  throw new Error('thrown from sync throw fail');
});

test('async skip pass', async (t) => {
  t.skip();
});

test('async pass', async () => {

});

test('async throw fail', async () => {
  throw new Error('thrown from async throw fail');
});

test('async skip fail', async (t) => {
  t.skip();
  throw new Error('thrown from async throw fail');
});

test('async assertion fail', async () => {
  // Make sure the assert module is handled.
  assert.strictEqual(true, false);
});

test('resolve pass', () => {
  return Promise.resolve();
});

test('reject fail', () => {
  return Promise.reject(new Error('rejected from reject fail'));
});

test('unhandled rejection - passes but warns', () => {
  Promise.reject(new Error('rejected from unhandled rejection fail'));
});

test('async unhandled rejection - passes but warns', async () => {
  Promise.reject(new Error('rejected from async unhandled rejection fail'));
});

test('immediate throw - passes but warns', () => {
  setImmediate(() => {
    throw new Error('thrown from immediate throw fail');
  });
});

test('immediate reject - passes but warns', () => {
  setImmediate(() => {
    Promise.reject(new Error('rejected from immediate reject fail'));
  });
});

test('immediate resolve pass', () => {
  return new Promise((resolve) => {
    setImmediate(() => {
      resolve();
    });
  });
});

test('subtest sync throw fail', async (t) => {
  await t.test('+sync throw fail', (t) => {
    t.diagnostic('this subtest should make its parent test fail');
    throw new Error('thrown from subtest sync throw fail');
  });
});

test('sync throw non-error fail', async (t) => {
  throw Symbol('thrown symbol from sync throw non-error fail')
});

test('level 0a', async (t) => {
  t.test('level 1a', async (t) => {
    const p1a = new Promise((resolve) => {
      setTimeout(() => {
        // console.log('resolving p1a');
        resolve();
      }, 1000);
    });

    return p1a;
  });

  t.test('level 1b', async (t) => {
    const p1b = new Promise((resolve) => {
      // console.log('resolving p1b');
      resolve();
    });

    return p1b;
  });

  t.test('level 1c', async (t) => {
    const p1c = new Promise((resolve) => {
      setTimeout(() => {
        // console.log('resolving p1c');
        resolve();
      }, 2000);
    });

    return p1c;
  });

  t.test('level 1d', async (t) => {
    const p1c = new Promise((resolve) => {
      setTimeout(() => {
        // console.log('resolving p1d');
        resolve();
      }, 1500);
    });

    return p1c;
  });

  const p0a = new Promise((resolve) => {
    setTimeout(() => {
      // console.log('resolving p0a');
      resolve();
    }, 3000);
  });

  return p0a;
});

test('top level', async (t) => {
  t.test('+long running', async (t) => {
    return new Promise((resolve, reject) => {
      setTimeout(resolve, 3000).unref();
      // setTimeout(resolve, 3000); //.unref();
    });
  });

  t.test('+short running', async (t) => {
    t.test('++short running', async (t) => {});
  });
});

test('invalid subtest - pass but subtest fails', (t) => {
  setImmediate(() => {
    t.test('invalid subtest fail', () => {
      throw new Error('this should not be thrown');
    });
  });
});
