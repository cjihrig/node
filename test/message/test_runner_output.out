TAP version 13
ok 1 - sync pass todo # TODO
  ---
  duration_ms: *
  ...
ok 2 - sync pass todo with message # TODO this is a passing todo
  ---
  duration_ms: *
  ...
not ok 3 - sync fail todo # TODO
  ---
  duration_ms: *
  error: thrown from sync fail todo
  failureType: testCodeFailure
  stack: |-
    *
    *
    *
    *
    *
    *
    *
    *
    *
    *
  ...
not ok 4 - sync fail todo with message # TODO this is a failing todo
  ---
  duration_ms: *
  error: thrown from sync fail todo with message
  failureType: testCodeFailure
  stack: |-
    *
    *
    *
    *
    *
    *
    *
    *
    *
    *
  ...
ok 5 - sync skip pass # SKIP
  ---
  duration_ms: *
  ...
ok 6 - sync skip pass with message # SKIP this is skipped
  ---
  duration_ms: *
  ...
ok 7 - sync pass
  ---
  duration_ms: *
  ...
# this test should pass
not ok 8 - sync throw fail
  ---
  duration_ms: *
  error: thrown from sync throw fail
  failureType: testCodeFailure
  stack: |-
    *
    *
    *
    *
    *
    *
    *
    *
    *
    *
  ...
ok 9 - async skip pass # SKIP
  ---
  duration_ms: *
  ...
ok 10 - async pass
  ---
  duration_ms: *
  ...
not ok 11 - async throw fail
  ---
  duration_ms: *
  error: thrown from async throw fail
  failureType: testCodeFailure
  stack: |-
    *
    *
    *
    *
    *
    *
    *
    *
    *
    *
  ...
not ok 12 - async skip fail # SKIP
  ---
  duration_ms: *
  error: thrown from async throw fail
  failureType: testCodeFailure
  stack: |-
    *
    *
    *
    *
    *
    *
    *
    *
    *
    *
  ...
not ok 13 - async assertion fail
  ---
  duration_ms: *
  error: Expected values to be strictly equal:

true !== false

  failureType: testCodeFailure
  stack: |-

    true !== false

    *
    *
    *
    *
    *
    *
    *
    *
    *
    *
  ...
ok 14 - resolve pass
  ---
  duration_ms: *
  ...
not ok 15 - reject fail
  ---
  duration_ms: *
  error: rejected from reject fail
  failureType: testCodeFailure
  stack: |-
    *
    *
    *
    *
    *
    *
    *
    *
    *
    *
  ...
ok 16 - unhandled rejection - passes but warns
  ---
  duration_ms: *
  ...
ok 17 - async unhandled rejection - passes but warns
  ---
  duration_ms: *
  ...
ok 18 - immediate throw - passes but warns
  ---
  duration_ms: *
  ...
ok 19 - immediate reject - passes but warns
  ---
  duration_ms: *
  ...
ok 20 - immediate resolve pass
  ---
  duration_ms: *
  ...
    not ok 1 - +sync throw fail
      ---
      duration_ms: *
      error: thrown from subtest sync throw fail
      failureType: testCodeFailure
      stack: |-
        *
        *
        *
        *
        *
        *
        *
        *
        *
        *
      ...
    # this subtest should make its parent test fail
    1..1
not ok 21 - subtest sync throw fail
  ---
  duration_ms: *
  error: '1 subtest failed'
  failureType: subtestsFailed
  ...
not ok 22 - sync throw non-error fail
  ---
  duration_ms: *
  error: Symbol(thrown symbol from sync throw non-error fail)
  failureType: testCodeFailure
  ...
    ok 1 - level 1a
      ---
      duration_ms: *
      ...
    ok 2 - level 1b
      ---
      duration_ms: *
      ...
    ok 3 - level 1c
      ---
      duration_ms: *
      ...
    ok 4 - level 1d
      ---
      duration_ms: *
      ...
    1..4
ok 23 - level 0a
  ---
  duration_ms: *
  ...
    not ok 1 - +long running
      ---
      duration_ms: *
      error: 'test did not finish before its parent and was cancelled'
      failureType: cancelledByParent
      ...
        ok 1 - ++short running
          ---
          duration_ms: *
          ...
        1..1
    ok 2 - +short running
      ---
      duration_ms: *
      ...
    1..2
not ok 24 - top level
  ---
  duration_ms: *
  error: '1 subtest failed'
  failureType: subtestsFailed
  ...
ok 25 - invalid subtest - pass but subtest fails
  ---
  duration_ms: *
  ...
not ok 26 - invalid subtest fail
  ---
  duration_ms: *
  error: 'test could not be started because its parent finished'
  failureType: parentAlreadyFinished
  stack: |-
    Immediate.<anonymous> (*)
  ...
1..26
# Warning: Test "unhandled rejection - passes but warns" generated asynchronous activity after the test ended. This activity created the error "Error: rejected from unhandled rejection fail" and would have caused the test to fail, but instead triggered an unhandledRejection event.
# Warning: Test "async unhandled rejection - passes but warns" generated asynchronous activity after the test ended. This activity created the error "Error: rejected from async unhandled rejection fail" and would have caused the test to fail, but instead triggered an unhandledRejection event.
# Warning: Test "immediate throw - passes but warns" generated asynchronous activity after the test ended. This activity created the error "Error: thrown from immediate throw fail" and would have caused the test to fail, but instead triggered an uncaughtException event.
# Warning: Test "immediate reject - passes but warns" generated asynchronous activity after the test ended. This activity created the error "Error: rejected from immediate reject fail" and would have caused the test to fail, but instead triggered an unhandledRejection event.
# tests 26
# pass 6
# fail 15
# skipped 3
# todo 2
# duration_ms *

